### ISSUE TITLE
Add  specialized serializer for Groovy

### ISSUE THREAD

**Issue by chaokunyang** [alipay/fury#684]

**Comment by chaokunyang** [comment]
Hi @azhsmesos, thank you for your willingness to contribute to Fury. Fury supports JVM classes serialization best if they don't custom serialization by JDK serialization method such as `writeReplace/writeObject/readResolve/readObject/readObjectNoData/writeExternal/readExternal`. The main work for optimizing groovy serialization are:
- find common groovy collections/containers/classes which needed to be serialized. Many groovy collections are collections in java.util.*, those collections are supported by fury already, you can just skip it.
- implement `io.fury.serializer.Serializer` for those classes in fury
- add a switch to register those serializers in fury

**Comment by chaokunyang** [comment]
For example, scala `object` is singleton, which defined `writeReplace` method, it return a `scala.runtime.ModuleSerializationProxy` for serialization. We should check whether groovy has similar behhaviour

**Comment by azhsmesos** [comment]
> 

Do we have a corresponding community similar to wechat group?  I want to go in to facilitate communication.

**Comment by chaokunyang** [comment]
> 

Yes, we do. You can follow our wechat offical acount https://github.com/alipay/fury#getting-involved  for joinning the group

**Comment by chaokunyang** [comment]
Java Serializers is enough for msot groovy types
